src/services/findAllReferences.ts

```diff
@@ -415,7 +415,11 @@ namespace ts.FindAllReferences.Core {
             return checker.getImmediateAliasedSymbol(symbol);
         }
 
-        return symbol;
+        // If the symbol is declared as part of a declaration like `{ type: "a" } | { type: "b" }`, use the property on the union type to get more references.
+        return symbol.declarations && firstDefined(symbol.declarations, decl =>

```

Comment:
getReferencedSymbolsForSymbol instead of returning propertyOfUnion, shouldnt you be returning all properties of the constituents with that name as symbols from ```getReferencedSymbolsForSymbol```

Reply:
If we use the property on the union as the search symbol, `populateSearchSymbolSet` will call `checker.getRootSymbols` to get the constituents of the union.

Reply:
Yeah makes sense.. Whats the reason to check if declaration is type node.. why not just check if the type at that location is union and return union property ?

Reply:
In: `{ type: "a" } | { type: "b" }`
If we're doing find-all-references on the first `type`, the declaration will be a PropertyDeclaration `type: "a"`. The parent is a TypeLiteralNode `{  type: "a" }` and parent.parent is the whole thing.
We have to go at least to `parent.parent` to get a union type node. If we make calls at any child of that we won't get a union type, but the nameless type for the node `{ type: "a" }` which is *not* a union, but a constituent of the union.